chromium_src/components/url_formatter/elide_url.cc

```diff
@@ -0,0 +1,26 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#define FormatOriginForSecurityDisplay \
+  FormatOriginForSecurityDisplay_ChromiumImpl
+#include "src/components/url_formatter/elide_url.cc"
+#undef FormatOriginForSecurityDisplay
+
+namespace url_formatter {
+
+std::u16string FormatOriginForSecurityDisplay(
+    const url::Origin& origin,
+    const SchemeDisplay scheme_display) {
+  auto display_origin =
+      FormatOriginForSecurityDisplay_ChromiumImpl(origin, scheme_display);
+  if (origin.scheme() == "chrome") {
```

Comment:
We need to restrict that replacement to `scheme_display == SchemeDisplay::SHOW`. Otherwise the first substring after the offset is not going to be the scheme.

For example, an `origin == "chrome://chrome-urls"` would give us `display_origin == "chrome-urls"` when `scheme_display != SchemeDisplay::SHOW` and this would become `brave-urls` which is not a scheme replacement.

Reply:
We can restrict it to `scheme_display == SchemeDisplay::SHOW`, but if `origin.scheme()` is `"chrome"`, the scheme gets prepended regardless of `scheme_display`'s value.

Reply:
The best would probably be to use `ShouldShowScheme()`:
```cpp
std::u16string FormatOriginForSecurityDisplay(
    const url::Origin& origin,
    const SchemeDisplay scheme_display) {
  auto display_origin =
      FormatOriginForSecurityDisplay_ChromiumImpl(origin, scheme_display);
  if (const auto& scheme = origin.scheme();
      scheme == "chrome" && ShouldShowScheme(scheme, scheme_display)) {
    base::ReplaceFirstSubstringAfterOffset(&display_origin, 0, u"chrome",
                                           u"brave");
  }

  return display_origin;
}
```

Reply:
> if `origin.scheme()` is `"chrome"`, the scheme gets prepended regardless of `scheme_display`'s value

Ah okay, so the value is ignored in that case. Thanks for the correction.

I think we may as well have the check, but it doesn't appear to be necessary right now.

Reply:
Actually, I started to like [this](https://github.com/brave/brave-core/pull/19561#discussion_r1283580109) better, as it automatically adjusts to any future changes in `ShouldShowScheme()`. WDYT @bridiver?